src/test/java/org/folio/permstest/TestUtil.java

```diff
@@ -55,43 +57,50 @@ public JsonObject getJson() {
   }
 
   public static Future<WrappedResponse> doRequest(Vertx vertx, String url,
-                                                  HttpMethod method, MultiMap headers, String payload,
-                                                  Integer expectedCode) {
+    HttpMethod method, MultiMap headers, String payload,
+    Integer expectedCode) {
     Promise<WrappedResponse> promise = Promise.promise();
-    boolean addPayLoad = false;
-    HttpClient client = vertx.createHttpClient();
-    HttpClientRequest request = client.requestAbs(method, url);
+    WebClient client = WebClient.create(vertx);
+    HttpRequest<Buffer> request = client.requestAbs(method, url);
     //Add standard headers
     request.putHeader("X-Okapi-Tenant", "diku")
-        .putHeader("content-type", CONTENT_TYPE_JSON)
-        .putHeader("accept", CONTENT_TYPE_JSON);
+      .putHeader("content-type", CONTENT_TYPE_JSON)
+      .putHeader("accept", CONTENT_TYPE_JSON);
     if (headers != null) {
-      for(Map.Entry entry : headers.entries()) {
-        request.putHeader((String)entry.getKey(), (String)entry.getValue());
+      for (Map.Entry entry : headers.entries()) {
+        request.putHeader((String) entry.getKey(), (String) entry.getValue());
       }
     }
-    //standard exception handler
-    request.exceptionHandler(e -> { promise.fail(e); });
-    request.handler( req -> {
-      req.bodyHandler(buf -> {
-        if(expectedCode != null && expectedCode != req.statusCode()) {
-          promise.fail(String.format("%s request to %s failed. Expected status code"
-                  + " '%s' but got status code '%s': %s", method, url,
-              expectedCode, req.statusCode(), buf));
-        } else {
-          System.out.println("Got status code " + req.statusCode() + " with payload of " + buf.toString());
-          WrappedResponse wr = new WrappedResponse(req.statusCode(), buf.toString(), req);
-          promise.complete(wr);
-        }
-      });
-    });
-    System.out.println("Sending " + method.toString() + " request to url '"+
-        url + " with payload: " + payload + "'\n");
+
+    System.out.println("Sending " + method.toString() + " request to url '" +
+      url + " with payload: " + payload + "'\n");
     if (method == HttpMethod.PUT || method == HttpMethod.POST) {
-      request.end(payload);
+      request.sendBuffer(Buffer.buffer(payload),
+        res -> handler(res, promise, expectedCode, method, url));
     } else {
-      request.end();
+      request.send(res -> handler(res, promise, expectedCode, method, url));
     }
     return promise.future();
   }
+
+  public static void handler(AsyncResult<HttpResponse<Buffer>> res,
+    Promise<WrappedResponse> promise, Integer expectedCode, HttpMethod method, String url) {
```

Comment:
```suggestion
  public static Future<WrappedResponse> handler(HttpResponse<Buffer> res,
    Integer expectedCode, HttpMethod method, String url) {
```
No need to create a Promise, you can directly return a Future.

Reply:
`handler` should be renamed, for example `assert`.